iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/SlackTerminalXmlSerializer.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.commons.extensions.AbstractExtensionXmlSerializer;
+import com.powsybl.commons.extensions.ExtensionXmlSerializer;
+import com.powsybl.commons.xml.XmlReaderContext;
+import com.powsybl.commons.xml.XmlWriterContext;
+import com.powsybl.iidm.network.*;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+@AutoService(ExtensionXmlSerializer.class)
+public class SlackTerminalXmlSerializer extends AbstractExtensionXmlSerializer<VoltageLevel, SlackTerminal> {
+
+    public SlackTerminalXmlSerializer() {
+        super("slackTerminal", "network", SlackTerminal.class, false, "slackTerminal.xsd",
+            "http://www.powsybl.org/schema/iidm/ext/slack_bus/1_0", "slt");
+    }
+
+    @Override
+    public void write(SlackTerminal slackTerminal, XmlWriterContext context) throws XMLStreamException {
+        writeTerminalRef(slackTerminal.getTerminal(), context.getExtensionsWriter());
+    }
+
+    @Override
+    public SlackTerminal read(VoltageLevel voltageLevel, XmlReaderContext context) {
+        Terminal terminal = readTerminalRef(voltageLevel.getNetwork(), context);
+        if (terminal != null) {
+            voltageLevel.newExtension(SlackTerminalAdder.class)
+                .setTerminal(terminal)
+                .add();
+            return voltageLevel.getExtension(SlackTerminalAdder.class);
+        } else {
+            return null;
+        }
+    }
+
+    public static void writeTerminalRef(Terminal t, XMLStreamWriter writer) throws XMLStreamException {
```

Comment:
I think this utility function already exists somewhere in iidm xml converter module.

Reply:
Indeed, in TerminalRefXml. A minor issue blocked the refactoring in the commit so far, but it is in progress!